src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1970,6 +1970,29 @@ internal MethodSymbol EnsureInlineArrayAsSpanExists(SyntaxNode syntaxNode, Named
                                                       diagnostics);
         }
 
+        internal NamedTypeSymbol EnsureInlineArrayTypeExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, int arrayLength, DiagnosticBag diagnostics)
+        {
+            Debug.Assert(Compilation.Assembly.RuntimeSupportsInlineArrayTypes);
+            Debug.Assert(arrayLength > 0);
+
+            string typeName = $"<>y__InlineArray{arrayLength}"; // see GeneratedNameKind.InlineArrayType
```

Comment:
>GeneratedNameKind.InlineArrayType [](http://example.com/codeflow?start=71&length=33)

Consider ` $"<>{GeneratedNameKind.InlineArrayType}__InlineArray{arrayLength}"` or adding an assertion `Debug.Assert("y" == GeneratedNameKind.InlineArrayType)` 

That way, GoToDefinition on `GeneratedNameKind.InlineArrayType` leads somewhere useful and that we don't accidentally modify it.